src/main/java/net/minecraftforge/fml/common/asm/transformers/SideTransformer.java

```diff
@@ -92,23 +92,26 @@
             }
         }
 
-        // remove dynamic lambda methods that are inside of removed methods
+        // remove dynamic synthetic lambda methods that are inside of removed methods
         List<Handle> dynamicLambdaHandles = lambdaGatherer.getDynamicLambdaHandles();
         if (!dynamicLambdaHandles.isEmpty())
         {
             methods = classNode.methods.iterator();
             while (methods.hasNext())
             {
                 MethodNode method = methods.next();
-                for (Handle dynamicLambdaHandle : dynamicLambdaHandles)
+                if ((method.access & Opcodes.ACC_SYNTHETIC) != 0)
```

Comment:
It's a commit adding only one line now.

Besides, I don't think that we need to check if it is referred only in sided methods, because the compiler seems to generate synthetic method respectively for every lambda having a lambda body, even though there are two lambdas that look exactly the same.